### DISCUSSION TITLE
Crossref with interactive tables (e.g., DT::datatable)

### DISCUSSION THREAD

**Discussion by jack-davison** [quarto-dev/quarto-cli#628] [Q&A] [answered]
Good afternoon,

I'm really enjoying using Quarto, so first a big thanks to the authors for putting it together! I'm especially enjoying the simplicity by which figures and tables can be labelled and cross-referenced, which brings me on to my query.

I'm a frequent user of the [DT](https://rstudio.github.io/DT/) package for interactive tables. An issue in RMarkdown was the automatic labelling of `DT::datatable()` outputs as "Figure X: ..." - see issue [here](https://github.com/rstudio/bookdown/issues/313). There were some workarounds, but I can't seem to recreate them in a .qmd file.

Does Quarto contain a more straightforward workaround for "forcing" an interactive table to be seen as a table? Under-the-hood in the HTML they may not strictly be tables, but from a reader's perspective they are, and it'd be strange to have to label them figures.

Cheers!

---

**Reproducible example**
```
```{r}
#| label: tbl-dt
#| tbl-cap: "My DT table."
DT::datatable(iris)
```    
Outputs:
![image](https://user-images.githubusercontent.com/45171616/163210817-452b6589-6dea-4ae4-a523-4e7185ff6874.png)

**Comment by cderv** [comment, marked as answer]
> An issue in RMarkdown was the automatic labelling of DT::datatable() outputs as "Figure X: ..."

Currently the limitation is the same as answered in https://github.com/quarto-dev/quarto-cli/discussions/782#discussioncomment-2669864. 
They can be referenced as figures 

````markdown
```{r}
#| label: fig-dt
#| tbl-cap: "My DT table."
DT::datatable(iris)
```
````

**DT** and other interactive tables are htmlwidgets and "special" HTML table, they cannot be referenced as Table yet. We'll try to improve support for this in the future. 

Related issues: https://github.com/quarto-dev/quarto-cli/issues/1084
